src/main/java/com/iota/iri/utils/log/StatusLogger.java

```diff
@@ -0,0 +1,133 @@
+ * updates about a certain task.
+ *
+ * Instead of printing all messages immediately and unnecessarily spamming the console output, it rate limits the amount
+ * of messages shown and only prints messages if the status has changed since the last output.
+ */
+public class StatusLogger {
+    /**
+     * Holds the interval in milliseconds that status messages get printed.
+     */
+    private static final int LOG_INTERVAL = 3000;
+
+    /**
+     * Holds a reference to the underlying logger.
+     */
+    private final Logger logger;
+
+    /**
+     * Holds the current status message that shall get printed to the console.
+     */
+    private String statusMessage = "";
+
+    /**
+     * Holds the last status message that was printed.
+     */
+    private String lastPrintedStatusMessage = "";
+
+    /**
+     * Holds a reference to the thread that takes care of scheduling the next status update.
+     */
+    private Thread outputThread = null;
+
+    /**
+     * Holds a timestamp value when the last message was printed.
+     */
+    private long lastLogTime = 0;
+
+    /**
+     * Constructor of the class.
+     *
+     * It simply stores the passed in parameters to be able to access them later on.
+     *
+     * @param logger {@link Logger} object that is used in combination with logback to issue messages in the console
+     */
+    public StatusLogger(Logger logger) {
+        this.logger = logger;
+    }
+
+    /**
+     * This method allows us to set a new status that shall get printed to the screen.
+     *
+     * It checks if the given message is new and then either prints the message immediately (if enough time has passed
+     * since the last message) or spawns a {@link Thread} that will print the new status after the given timeout.
+     *
+     * When creating the {@link Thread} it copies its name so the output is "transparent" to the user and the effect
+     * is the same as dumping the message manually through the {@link Logger} object itself.
+     *
+     * @param message status message that shall get printed
+     */
+    public void status(String message) {
+        if (!message.equals(statusMessage)) {
+            statusMessage = message;
+
+            if (System.currentTimeMillis() - lastLogTime >= LOG_INTERVAL) {
+                printStatusMessage();
+            } else if (outputThread == null) {
+                synchronized(this) {
+                    if (outputThread == null) {
+                        outputThread = new Thread(() -> {
+                            try {
+                                Thread.sleep(Math.max(LOG_INTERVAL - (System.currentTimeMillis() - lastLogTime), 1));
+                            } catch(InterruptedException e) { /* do nothing */ }
+
+                            printStatusMessage();
+
+                            outputThread = null;
+                        }, Thread.currentThread().getName());
```

Comment:
So you give the `outputThread` the same name of the thread that happened to call the `status` method??

Reply:
Anyhows, just for readability i would wrap the logic of creating the thread in a private method.